loopdb/sqlite.go

```diff
@@ -219,3 +310,70 @@ func NewSqlReadOpts() *SqliteTxOptions {
 func (r *SqliteTxOptions) ReadOnly() bool {
 	return r.readOnly
 }
+
+// parseSqliteTimeStamp parses a timestamp string in the format of
+// "YYYY-MM-DD HH:MM:SS +0000 UTC" and returns a time.Time value.
+// NOTE: we can't use time.Parse() because it doesn't support having years
+// with more than 4 digits.
+func parseSqliteTimeStamp(dateTimeStr string) time.Time {
+	// Split the date and time parts.
+	parts := strings.Fields(strings.TrimSpace(dateTimeStr))
+	datePart, timePart := parts[0], parts[1]
+
+	// Parse the date.
+	dateParts := strings.Split(datePart, "-")
+	year, _ := strconv.Atoi(dateParts[0])
+	month, _ := strconv.Atoi(dateParts[1])
+	day, _ := strconv.Atoi(dateParts[2])
+
+	// Parse the time.
+	timeParts := strings.Split(timePart, ":")
+	hour, _ := strconv.Atoi(timeParts[0])
+	minute, _ := strconv.Atoi(timeParts[1])
+	second, _ := strconv.Atoi(timeParts[2])
+
+	// Construct a time.Time value.
+	t := time.Date(year, time.Month(month), day, hour, minute, second, 0, time.UTC)
+
+	return t
+}
+
+// parseSqliteTimeStamp parses a timestamp string in the format of
+// "YYYY-MM-DDTHH:MM:SSZ" and returns a time.Time value.
+// NOTE: we can't use time.Parse() because it doesn't support having years
+// with more than 4 digits.
+func parsePostgresTimeStamp(dateTimeStr string) time.Time {
+	// Split the date and time parts.
+	parts := strings.Split(dateTimeStr, "T")
+	datePart, timePart := parts[0], strings.TrimSuffix(parts[1], "Z")
+
+	// Parse the date.
```

Comment:
The rest of this function is the same as for the sqlite case. Perhaps makes sense to extract to a separate function and also add the error handling in so any malformed stings wont bite us.